src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java

```diff
@@ -361,33 +365,34 @@ private Map<Long, Long> importTasks(final ProgrammingExercise templateExercise,
             programmingExerciseTaskRepository.save(copy);
             newIdByOldId.put(task.getId(), copy.getId());
             return copy;
-        }).collect(Collectors.toSet()));
-
+        }).collect(Collectors.toList()));
```

Comment:
```suggestion
        }).toList());
```

Reply:
This is actually not possible.
`toList()` returns an unmodifiable List while `collect(Collectors.toList()))` returns an ArrayList. Since this list is used in a to-be-saved exercise Hibernate will try to replace the items in this list with the newly saved ones. 
Therefore we need the modifiable list here.